src/providers/r2Provider.ts

```diff
@@ -103,18 +95,21 @@ function r2MetadataToHeaders(
 
   return {
     etag: object.httpEtag,
+    'content-type':
+      object.httpMetadata?.contentType ?? 'application/octet-stream',
     'accept-range': 'bytes',
     // https://github.com/nodejs/build/blob/e3df25d6a23f033db317a53ab1e904c953ba1f00/ansible/www-standalone/resources/config/nodejs.org?plain=1#L194-L196
     'access-control-allow-origin': object.key.endsWith('.json')
       ? '*'
       : undefined,
     'cache-control':
       httpStatusCode === 200 ? CACHE_HEADERS.success : CACHE_HEADERS.failure,
-    expires: httpMetadata?.cacheExpiry?.toUTCString(),
+    expires: httpMetadata?.cacheExpiry?.toUTCString() ?? '',
```

Comment:
Are these headers necessary to be empty strings? I'm not sure an empty string is a valid `expires` header; Afaik if not defined, these values should be undefined, which would omit these headers.

Reply:
Unfortunately it is neccesary, the types only allow for `string`, and omitting the `?? ''` here would have the header sent as `expires: undefined` when `cacheExpiry` is undefined. Empty headers are omitted from the response sent to the client however

I could separate the headers from the fetch call which might be able to clean it up, something like

```typescript
const headers = new Headers({
  /* headers that always exist */
})
if (httpMetadata?.cacheExpiry) {
  headers.set('expires', httpMetadata.cacheExpiry.toUTCString();
}
// ...
```